### PR TITLE
Revise "batch-norm_vn" - Phần 3

### LINKED ISSUE #1556: Revise "batch-norm_vn" - Phần 3
Phần này được dịch bởi:
* **dacdinh147**
* **tranyenthy**

Nếu bạn đã dịch phần này, vui lòng bỏ qua việc revise.

### PR DESCRIPTION
**Pull request by cuongvng**
Close #1556

### CODE REVIEW

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 1]
```diff
@@ -350,11 +350,11 @@ The `num_features` parameter required by the `BatchNorm` instance is the number
 The `num_dims` parameter also required by this instance is 2 for a fully-connected layer and 4 for a convolutional layer.
 -->
 
-Bây giờ chúng ta có thể tạo ra một tầng `BatchNorm` đúng cách.
+Giờ ta có thể tạo một tầng `BatchNorm` đúng cách.
 Tầng này sẽ duy trì những tham số thích hợp tương ứng với tỉ lệ `gamma` và độ chệch `beta`, hai tham số này sẽ được cập nhật trong quá trình huấn luyện.
-Thêm vào đó, tầng BN sẽ sẽ duy trì một giá trị trung bình động của những giá trị trung bình và phương sai cho các lần chạy sau trong lúc mô hình dự đoán. 
-Tham số `num_features` được yêu cầu truyền vào bởi đối tượng `BatchNorm` là số lượng các đầu ra cho tầng kết nối đầy đủ và số lượng kênh đầu ra cho tầng tích chập.
-Tham số `num_dims` cũng được yêu cầu truyền vào bởi đối tượng này là 2 cho tầng kết nối đầy đủ và 4 cho tầng tích chập.
+Thêm vào đó, tầng BN sẽ duy trì giá trị trung bình động của trung bình và phương sai sử dụng về sau khi ở chế độ dự đoán.
```
Comment (rootonchair) [inline chapter_convolutional-modern/batch-norm_vn.md:355]:
```suggestion
Thêm vào đó, tầng BN sẽ duy trì giá trị trung bình động của trung bình và phương sai để sử dụng về sau khi ở chế độ dự đoán. 
```

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 2]
```diff
@@ -371,13 +371,13 @@ Also note that for the sake of convenience we did not worry about automatically
...
-Sau đó, ta tích hợp chức năng này vào một tầng tùy chỉnh, với mã nguồn chủ yếu để giải quyết các vấn đề quản trị,
-chẳng hạn như di chuyển dữ liệu đến thiết bị phù hợp ngữ cảnh, cấp phát và khởi tạo bất kỳ biến nào được yêu cầu, theo dõi các giá trị trung bình động (của trung bình và phương sai trong trường hợp này), v.v.
-Khuôn mẫu này cho phép ta tách hoàn toàn các phép tính toán ra khỏi đoạn mã phụ trợ.
-Cũng lưu ý rằng để thuận tiện, ta không cần lo về việc tự động suy ra kích thước đầu vào ở đây, do đó chúng ta cũng không cần chỉ định số lượng đặc trưng xuyên suốt.
-Đừng lo lắng, lớp `BatchNorm` của Gluon sẽ làm điều này.
+Tạm để thuật toán sang một bên và tập trung vào khuôn mẫu thiết kế (*design pattern*) của việc lập trình. 
+Thông thường, ta lập trình phần toán trong một hàm riêng biệt, ví dụ như `batch_norm`.
+Sau đó, ta tích hợp chức năng này vào một tầng tùy chỉnh, với mã nguồn chủ yếu giải quyết các vấn đề phụ trợ,
+như di chuyển dữ liệu đến thiết bị phù hợp, cấp phát và khởi tạo biến, theo dõi các giá trị trung bình động (của trung bình và phương sai trong trường hợp này), v.v.
```
Comment (rootonchair) [inline chapter_convolutional-modern/batch-norm_vn.md:377]:
```suggestion
Sau đó, ta tích hợp chức năng này vào một tầng tùy chỉnh, với mã nguồn chủ yếu giải quyết các vấn đề phụ trợ như di chuyển dữ liệu đến thiết bị phù hợp, cấp phát và khởi tạo biến, theo dõi các giá trị trung bình động (của trung bình và phương sai trong trường hợp này), v.v.
```

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 3]
```diff
@@ -371,13 +371,13 @@ Also note that for the sake of convenience we did not worry about automatically
...
+Khuôn mẫu này giúp tách biệt việc tính toán khỏi các đoạn mã dài.
```
Comment (rootonchair) [inline chapter_convolutional-modern/batch-norm_vn.md:378]:
```suggestion
Khuôn mẫu này giúp tách biệt việc tính toán khỏi các đoạn mã rập khuôn.
```
Dịch như vầy có chuẩn hơn không nhỉ?

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 4]
```diff
@@ -371,13 +371,13 @@ Also note that for the sake of convenience we did not worry about automatically
...
+Cũng lưu ý rằng để thuận tiện, khi lập trình BN từ đầu, ta không tự động suy ra kích thước đầu vào, do đó cần chỉ định số lượng đặc trưng xuyên suốt.
```
Comment (rootonchair) [inline chapter_convolutional-modern/batch-norm_vn.md:379]: Câu này hơi khó. Hóng cao nhân

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 5]
```diff
@@ -413,15 +413,15 @@ class BatchNorm(nn.Block):
 ## Using a Batch Normalization LeNet
 -->
 
-## Sử dụng một cấu trúc mạng LeNet Chuẩn hóa theo Batch 
+## Sử dụng LeNet có BN
```
Comment (rootonchair) [inline chapter_convolutional-modern/batch-norm_vn.md:416]:
```suggestion
## Sử dụng LeNet với Chuẩn hóa theo Batch
```
Viết tắt ở header mình thấy hơi kỳ

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 6]
```diff
@@ -413,15 +413,15 @@ class BatchNorm(nn.Block):
...
 
 <!--
 To see how to apply `BatchNorm` in context, below we apply it to a traditional LeNet model (:numref:`sec_lenet`).
 Recall that BN is typically applied after the convolutional layers and fully-connected layers but before the corresponding activation functions.
 -->
 
-Để xem cách áp dụng `BatchNorm` trong ngữ cảnh, bên dưới chúng tôi áp dụng nó cho mô hình LeNet truyền thống (:numref:`sec_lenet`).
-Hãy nhớ lại rằng BN thường được áp dụng sau các tầng tích chập và các lớp được kết nối đầy đủ nhưng trước các hàm kích hoạt tương ứng.
+Để biết cách áp dụng `BatchNorm` trên thực tế, bên dưới ta áp dụng cho mô hình LeNet truyền thống (:numref:`sec_lenet`).
+Nhớ lại rằng BN thường được áp dụng sau tầng tích chập và tầng kết nối đầy đủ và trước hàm kích hoạt tương ứng.
```
Comment (rootonchair) [inline chapter_convolutional-modern/batch-norm_vn.md:424]:
```suggestion
Nhắc lại rằng BN thường được sử dụng sau tầng tích chập và tầng kết nối đầy đủ và trước hàm kích hoạt tương ứng.
```

File: chapter_convolutional-modern/batch-norm_vn.md (hunk 3 above)
Comment (cuongvng) [reply chapter_convolutional-modern/batch-norm_vn.md:377]: yeah mình đồng ý!

File: chapter_convolutional-modern/batch-norm_vn.md (hunk 4 above)
Comment (cuongvng) [reply chapter_convolutional-modern/batch-norm_vn.md:378]: Câu này match với phần revise sau, có 1 đoạn so sánh BN tự lập trình với BN của Gluon. BN của Gluon dùng defered initialization, còn BN tự lập trình thì ko.
>Compared with the `BatchNorm` class, which we just defined ourselves, the `BatchNorm` class defined by the `nn` model in Gluon is easier to use.
In Gluon, we do not have to worry about `num_features` or `num_dims`.
Instead, these parameter values will be inferred automatically via delayed initialization.

Comment (rootonchair) [reply chapter_convolutional-modern/batch-norm_vn.md:378]: Ahhh. Vậy thì hợp lý rồi. Thanks!

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 7]
```diff
@@ -297,13 +297,13 @@ Hãy nhớ rằng dropout cũng thể hiện tính chất này.
...
 
-## Lập trình Từ đầu
+## Lập trình từ đầu
 
 <!--
 Below, we implement a batch normalization layer with `ndarray`s from scratch:
 -->
 
-Dưới đây, chúng ta lập trình lại từ đầu tầng chuẩn hoá theo batch mà chỉ dùng $ndarrays$.
+Dưới đây, chúng ta lập trình tầng chuẩn hoá theo batch chỉ dùng $ndarrays$.
```
Comment (duythanhvn) [inline chapter_convolutional-modern/batch-norm_vn.md:306]:
```suggestion
Dưới đây, chúng ta lập trình tầng chuẩn hoá theo batch chỉ dùng `ndarray`.
```

File: chapter_convolutional-modern/batch-norm_vn.md [hunk 8]
```diff
@@ -371,13 +371,12 @@ Also note that for the sake of convenience we did not worry about automatically
...
-Khuôn mẫu này cho phép ta tách hoàn toàn các phép tính toán ra khỏi đoạn mã phụ trợ.
-Cũng lưu ý rằng để thuận tiện, ta không cần lo về việc tự động suy ra kích thước đầu vào ở đây, do đó chúng ta cũng không cần chỉ định số lượng đặc trưng xuyên suốt.
-Đừng lo lắng, lớp `BatchNorm` của Gluon sẽ làm điều này.
+Tạm để thuật toán sang một bên và tập trung vào khuôn mẫu thiết kế (*design pattern*) của việc lập trình. 
+Thông thường, ta lập trình phần toán trong một hàm riêng biệt, ví dụ như `batch_norm`.
+Sau đó, ta tích hợp chức năng này vào một tầng tùy chỉnh, với mã nguồn chủ yếu giải quyết các vấn đề phụ trợ như di chuyển dữ liệu đến thiết bị phù hợp, cấp phát và khởi tạo biến, theo dõi các giá trị trung bình động (của trung bình và phương sai trong trường hợp này), v.v.
+Khuôn mẫu này giúp tách biệt việc tính toán khỏi các đoạn mã rập khuôn.
+Cũng lưu ý rằng để thuận tiện, khi lập trình BN từ đầu, ta không tự động suy ra kích thước đầu vào, do đó cần chỉ định số lượng đặc trưng xuyên suốt.
```
Comment (duythanhvn) [inline chapter_convolutional-modern/batch-norm_vn.md:378]:
```suggestion
Cũng lưu ý rằng để thuận tiện khi lập trình BN từ đầu, ta không tự động suy ra kích thước đầu vào, do đó ta cần chỉ định số lượng đặc trưng xuyên suốt.
```
